source/clear-linux/guides/maintenance/kernel-modules-dkms.rst

```diff
@@ -0,0 +1,307 @@
+.. _kernel-modules-dkms:
+
+Add kernel modules with DKMS
+############################
+
+Kernel modules are additional pieces of software capable of being inserted 
+into the Linux kernel to add functionality, such as a hardware driver. 
+Kernel modules may already be part of the Linux source tree (in-tree) or may 
+come from an external source, such as directly from a vendor (out-of-tree).  
+
+In cases where drivers beyond those enabled by default in |CL-ATTR| are
+needed it may be necessary to manually build out-of-tree modules. 
+
+Out-of-tree kernel modules can be can be  `manually built and maintained
+<kernel-modules>`_. Out-of-tree kernel modules can also be managed with the
+`Dynamic Kernel Module System (DKMS)`_ on |CL| using the instructions in this
+document.
+
+:abbr:`DKMS (Dynamic Kernel Module System)` is a framework that facilitates
+the building and installation of kernel modules. This allows |CL| to provide
+hooks that automatically rebuild modules against new kernel versions. 
+
+
+.. contents:: :local:
+   :depth: 1
+   :backlinks: top
+
+
+
+.. include:: kernel-modules.rst
+   :start-after: kernel-modules-availability-begin:
+   :end-before: kernel-modules-availability-end:
+
+
+
+
+Install DKMS on |CL|
+====================
+
+The *kernel-native-dkms* bundle provides the :command:`dkms`: program and
+Linux kernel headers, which are required for compiling kernel modules.
+
+The *kernel-native-dkms* bundle also:
+
+* Adds a systemd update trigger
+  (:file:`/usr/lib/systemd/system/dkms-new-kernel.service`) to automatically
+  run DKMS to rebuild modules after a kernel upgrade occurs with :ref:`swupd
+  update <swupd-guide>`.
+
+* Disables kernel modules signature verification by appending a kernel
+  command-line parameter (:command:`module.sig_unenforce`) from
+  :file:`/usr/share/kernel/cmdline.d/clr-ignore-mod-sig.conf`.
+
+* Adds a notification to the Message of the Day (MOTD) indicating kernel
+  modules signature verification is disabled. 
+
+.. warning::
+
+   #. It is important to always review the output of :command:`swupd update` to
+      make sure kernel modules rebuilt against the new kernel successfully. This is
+      especially important for systems where a successful boot relies on a kernel
+      module.
+
+   #. Kernel modules installed with DKMS typically put source and binary files
+      under the :file:`/usr` subdirectory. This conflicts with the :ref:`stateless
+      <stateless>` design principles of |CL| and the :command:`swupd verify
+      --fix` command should be avoided with out-of-tree kernel modules in use.   
+
+Install the *kernel-native-dkms* bundle:
+
+#. From a |CL| system, ensure you are running the *native* kernel. 
+   Currently only the native kernel is enabled to build and load
```

Comment:
We now also have kernel-lts-dkms, which is probably even more useful since things like the Nvidia drivers don't build against the latest kernel yet. I may need to fix that bundle today, though.